topics/user_interface_components/status_bar_widgets.md

```diff
@@ -0,0 +1,114 @@
+[//]: # (title: Status Bar Widgets)
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+
+The IntelliJ Platform allows plugins to extend the IDE [status bar](https://www.jetbrains.com/help/idea/guided-tour-around-the-user-interface.html#status-bar) with additional custom widgets.
+
+Status bar widgets are small UI elements that allow providing users with useful information and settings for the current file, project, IDE, etc.
+For example, the status bar contains the widget showing the encoding of the current file, or the current VCS branch of the project.
+
+The starting point for extending the status bar with new widgets is the
+[`StatusBarWidgetFactory`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/StatusBarWidgetFactory.java)
+interface, which is registered in the `com.intellij.statusBarWidgetFactory` extension point.
+
+In case a widget provides information or functionality related to the editor files, consider extending the
+[`StatusBarEditorBasedWidgetFactory`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/impl/status/widget/StatusBarEditorBasedWidgetFactory.java)
+class.
+
+Each widget factory returns a new widget from `createWidget()`.
+To control the disposing of a widget, implement the `disposeWidget()`, if you just want to dispose it, use `Disposer.dispose(widget)`.
+
+Any widget should implement the
+[`StatusBarWidget`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/StatusBarWidget.java)
+interface.
+
+To reuse the IntelliJ Platform implementation, you can extend one of two classes:
+
+- [`EditorBasedWidget`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/impl/status/EditorBasedWidget.java)
+- [`EditorBasedStatusBarPopup`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/impl/status/EditorBasedStatusBarPopup.java)
+
+## EditorBasedWidget
+
+`EditorBasedWidget` is the basic widget implementation.
+To implement it, override `ID()` where returns the unique ID of the widget.
+This identifier may be needed to later get a widget instance.
```

Comment:
get -> obtain

can we add a short description of use-case? maybe something like "to later obtain a widget instance, e.g. to update its status" or similar

Reply:
Fixed.